rotkehlchen/chain/evm/decoding/thegraph/balances.py

```diff
@@ -40,6 +45,59 @@ def __init__(
         self.token = native_asset.resolve_to_evm_token()
         self.staking_contract = staking_contract
 
+    def _query_vesting_balances(self, balances: 'BalancesSheetType') -> 'BalancesSheetType':
+        """Query balances of vested GRT tokens if DelegationTransfer events are found."""
+        db_filter = EvmEventFilterQuery.make(
+            counterparties=[self.counterparty],
+            location=Location.ETHEREUM,
+            event_types=[HistoryEventType.INFORMATIONAL],
+            event_subtypes=[HistoryEventSubType.NONE],
+        )
+        with self.event_db.db.conn.read_ctx() as cursor:
+            events = self.event_db.get_history_events(
+                cursor=cursor,
+                filter_query=db_filter,
+                has_premium=True,
+            )
+        vesting_contracts = defaultdict(list)
+        for event in events:
+            if (
+                not event.location_label or
+                not event.extra_data or
+                not (delegator := event.extra_data.get('delegator_l2'))
+            ):
+                continue
+            vesting_contracts[string_to_evm_address(event.location_label)].append(delegator)
+
+        if len(vesting_contracts) == 0:
+            return balances
+
+        grt_price = Inquirer.find_usd_price(self.token)
+        for address, delegators in vesting_contracts.items():
+            for delegator in delegators:
+                if self.evm_inquirer.get_code(delegator) != '0x':  # if contract
+                    try:
+                        vested_balance = self.evm_inquirer.call_contract(
+                            contract_address=delegator,  # the vesting contract
+                            abi=GRAPH_TOKEN_LOCK_WALLET_ABI,
+                            method_name='totalOutstandingAmount',
+                        )
```

Comment:
instead of doing this in a for loop create a multicall (I believe multicall_2). If something it is not a contract it will fail and you can log and continue. multicall_2 returns for each call a boolean and the output. boolean true if it succeded and false otherwise